lib/screens/course_schedule.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:concordia_navigation/models/course.dart';
+import 'package:concordia_navigation/providers/calendar_data.dart';
+import 'package:concordia_navigation/widgets/weekday.dart';
+import 'package:flutter/material.dart';
+import 'package:concordia_navigation/services/localization.dart';
+import 'package:provider/provider.dart';
+
+/// User Schedule Screen
+/// This should have no business logic, only handling data generated by service classes and models.
+class CourseSchedule extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    List<Iterable<Course>> weekdays =
+        Provider.of<CalendarData>(context).schedule?.byWeekday();
+    List<Widget> weekdayContainer = [];
+
+    while (weekdays == null) {
+      return Center(child: CircularProgressIndicator());
+    }
+
+    for (int i = 0; i < weekdays.length; i++) {
```

Comment:
Can you explain what this is doing? Maybe add a comment

Reply:
resolved in 8e68243097e26955347c9ccb6014071136c7cbe9